packages/amplify-provider-awscloudformation/src/root-stack-builder/root-stack-builder.ts

```diff
@@ -0,0 +1,252 @@
+    cfnOutputs.deploymentBucketName = new cdk.CfnOutput(this, 'DeploymentBucketName', {
+      description: 'CloudFormation provider root stack deployment bucket name',
+      value: cdk.Fn.ref('DeploymentBucketName'),
+      exportName: cdk.Fn.sub('${AWS::StackName}-DeploymentBucketName'),
+    });
+
+    cfnOutputs.authRoleName = new cdk.CfnOutput(this, 'AuthRoleName', {
+      value: cdk.Fn.ref('AuthRoleName'),
+    });
+
+    cfnOutputs.unauthRoleName = new cdk.CfnOutput(this, 'UnauthRoleName', {
+      value: cdk.Fn.ref('UnauthRoleName'),
+    });
+    this.templateObj = {};
+    this.templateObj['Outputs'] = { ...cfnOutputs };
+  }
+
+  toCloudFormation() {
+    prepareApp(this);
+    return this._toCloudFormation();
+  }
+}
+
+export interface ResourceConfig {
+  resourceName?: string;
+  category: string;
+  stackFileName: string;
+}
+
+export interface RootStackTransformOptions {
+  //inputParser: IInputParser[];  // not needed for root stack
+  resourceConfig?: ResourceConfig;
+}
+
+export interface DeploymentOptions {
+  templateStack: Template;
+}
+
+export class AmplifyRootStackTransform {
+  private _rootTemplateObj: AmplifyRootStackTemplateProps; // Props to modify Root stack data
+  private _resourceConfig: ResourceConfig; // Config about resource to build
+  private _rootStackOptions: RootStackOptions; // options to help generate  cfn template
+  private _command: CommandType;
+  constructor(options: RootStackTransformOptions, command: CommandType) {
+    this._resourceConfig = options.resourceConfig;
+    this._command = command;
+  }
+
+  public async transform(): Promise<Template> {
+    // parse Input data
+    this._rootStackOptions = await this.getInput(); // get RootStackOPtions from cli-inputs.json
+    // generate cfn stack
+    const template = await this.generateRootStackTemplate();
+    // save stack
+    if (this._command === CommandType.PUSH) {
+      await this.deployOverrideStacksToDisk({
+        templateStack: template,
+      });
+    }
+    return template;
+  }
+  /**
+   *
+   * @returns Object required to generate Stack using cdk
+   */
+  private getInput = async (): Promise<RootStackOptions> => {
+    if (this._command === CommandType.INIT) {
+      const buildConfig: RootStackOptions = {
+        event: this._command,
+        rootStackFileName: this._resourceConfig.stackFileName,
+      };
+      return buildConfig;
+    } else {
+      const projectPath = pathManager.findProjectRoot();
+      const buildConfig: RootStackOptions = {
+        event: this._command,
+        rootStackFileName: this._resourceConfig.stackFileName,
+        overrideDir: pathManager.getOverrideDirPath(projectPath, 'root'),
+      };
+      return buildConfig;
+    }
+  };
+
+  /**
+   * Generates Root stack Template
+   * @returns CFN Template
+   */
+  private generateRootStackTemplate = async (): Promise<Template> => {
+    const stack = new AmplifyRootStack(undefined as any, 'Amplify');
```

Comment:
passing in `undefined` does not seems to be ok here as breaks the CDK graph and purpose. Can't we do it like we do in transformer V2, a CDK supported way via custom CdkApp and Custom Synthesizer?